libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -445,7 +442,12 @@ void AP_Frsky_Telem::send_D(void)
     // send frame2 every second
     if (now - _D.last_1000ms_frame >= 1000) {
         _D.last_1000ms_frame = now;
-        send_uint16(DATA_ID_GPS_COURS_BP, (uint16_t)((_ahrs.yaw_sensor / 100) % 360)); // send heading in degree based on AHRS and not GPS
+        {
+            AP_AHRS &_ahrs = AP::ahrs();
+            WITH_SEMAPHORE(_ahrs.get_semaphore());
+            yaw_sensor = _ahrs.yaw_sensor;
```

Comment:
accessing a single integer is atomic, no need to take a semaphore

Reply:
ok, how about floats? since floats are 4 byte aligned as well and in the osd code [here](https://github.com/ArduPilot/ardupilot/blob/899335e30b3ba627a0049a03ca3577dd6253e35b/libraries/AP_OSD/AP_OSD_Screen.cpp#L1092) a semaphore is held to access them I did the same

Reply:
floats and integers should be both atomic, I removed the semaphores in accessing them